cmd/cookbook_versions_cmd.go

```diff
@@ -41,20 +41,20 @@ func CookbookVersionList(c *cli.Context) error {
 	}
 
 	labelables := make([]types.Labelable, len(cookbookVersions))
-	for i:=0; i< len(cookbookVersions); i++ {
-		labelables[i] = types.Labelable(&cookbookVersions[i])
+	for i := 0; i < len(cookbookVersions); i++ {
+		labelables[i] = types.Labelable(cookbookVersions[i])
 	}
 	labelIDsByName, labelNamesByID := LabelLoadsMapping(c)
 	filteredLabelables := LabelFiltering(c, labelables, labelIDsByName)
 	LabelAssignNamesForIDs(c, filteredLabelables, labelNamesByID)
-	cookbookVersions = make([]types.CookbookVersion, len(filteredLabelables))
-	for i, labelable := range filteredLabelables {
+	cookbookVersions = make([]*types.CookbookVersion, len(filteredLabelables))
+	for _, labelable := range filteredLabelables {
 		s, ok := labelable.(*types.CookbookVersion)
 		if !ok {
 			formatter.PrintFatal("Label filtering returned unexpected result",
 				fmt.Errorf("expected labelable to be a *types.CookbookVersion, got a %T", labelable))
 		}
-		cookbookVersions[i] = *s
+		cookbookVersions = append(cookbookVersions, s)
```

Comment:
With this change you end up with a cookbookVersions slice whose length is double the one in filteredLabelables, with the actual labelables at the end, preceded by len(filteredLabelables) nil values. This is repeated in the list commands for other Labelables

Reply:
Good catch! Done!